packages/opal-client/opal_client/data/updater.py

```diff
@@ -438,3 +370,107 @@ async def update_policy_data(
                     whole_report, extra_callbacks
                 )
             )
+
+    async def handle_delete_data(self, store_transaction, entry):
+        policy_store_path = "" if entry is None else entry.dst_path
+        logger.info(
+            "Deleting data in policy-store: destination path='{path}'",
+            path=policy_store_path or "/",
+        )
+        try:
+            await store_transaction.set_policy_data(
+                method="DELETE",
+                policy_data=None,
+                path=policy_store_path
+            )
+        except Exception:
+            logger.exception(
+                "Failed to delete data in policy-store")
+            # re-raise so the context manager will be aware of the failure
+            raise
+
+    async def handle_new_data(self, reports, store_transaction, url, result, entry):
+        error_content = None
+        fetched_data_successfully = True
+
+        if isinstance(result, Exception):
+            fetched_data_successfully = False
+            logger.error(
+                "Failed to fetch url {url}, got exception: {exc}",
+                url=url,
+                exc=result,
+            )
+        if isinstance(
+            result, aiohttp.ClientResponse
+        ) and is_http_error_response(
+            result
+        ):  # error responses
+            fetched_data_successfully = False
+            try:
+                error_content = await result.json()
+                logger.error(
+                    "Failed to fetch url {url}, got response code {status} with error: {error}",
+                    url=url,
+                    status=result.status,
+                    error=error_content,
+                )
+            except json.JSONDecodeError:
+                error_content = await result.text()
+                logger.error(
+                    "Failed to decode response from url:{url}, got response code {status} with response: {error}",
+                    url=url,
+                    status=result.status,
+                    error=error_content,
+                )
+        store_transaction._update_remote_status(
+            url=url, status=fetched_data_successfully, error=str(
+                error_content)
+        )
+
+        if fetched_data_successfully:
+            # get path to store the URL data (default mode (None) is as "" - i.e. as all the data at root)
+            policy_store_path = "" if entry is None else entry.dst_path
+            # always default to PUT
+            method = "PUT" if entry is None else entry.save_method
```

Comment:
You already have "PUT" as the default method in `set_policy_data`,
So it might be better to not pass it at all when it's none and let each policy store (there would be other stores in OPAL soon) have its own defaults.